lua/competitest/receive.lua

```diff
@@ -110,8 +110,10 @@ end
 ---@param single_file_format string: string with CompetiTest modifiers to match single testcases file name
 ---@param input_file_format string: string with CompetiTest modifiers to match input files name
 ---@param output_file_format string: string with CompetiTest modifiers to match output files name
-function M.store_problem(filepath, tcdir, tclist, use_single_file, single_file_format, input_file_format, output_file_format)
-	utils.write_string_on_file(filepath, "")
+---@param template_file string: string with the absolute location of template file to use.
+function M.store_problem(filepath, tcdir, tclist, use_single_file, single_file_format, input_file_format, output_file_format, template_file)
+	local template = utils.load_file_as_string(vim.fs.normalize(utils.eval_string(filepath, template_file, nil)))
+	utils.write_string_on_file(filepath, template)
```

Comment:
Instead of loading and then writing, just use `luv.fs_copyfile()`.

Since `vim.fs` isn't available in Neovim 0.5 please use `luv.fs_realpath()` instead of `vim.fs.normalize`. At the moment Neovim 0.5 is still supported by CompetiTest, but soon or later I'll bump minimum required version to 0.7.

Reply:
I was unable to find what `luv.fs_realpath()` does. I ran
`:lua = vim.loop.fs_realpath("~/.config/")` which gave no such file.

I am using latest neovim version, if this function is dropped in new versions, are there any alternatives?

Are there any plans for upgrading to Neovim 0.7?

Reply:
Sorry I've forgot `luv.fs_realpath()` only expands `.` and `..` in a path.
At the moment we can expand tilde with gsub:
``` lua
filepath = string.gsub(filepath, "^%~", luv.os_homedir())
```

Reply:
Solved with 445e861